### PR TITLE
Header

### LINKED ISSUE #2: Header content & styling

### PR DESCRIPTION
**Pull request by niekraaijmakers**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #2 

Test URLs:
- Before: https://main--nokia--hlxsites.hlx.page/
- After:  https://header--nokia--hlxsites.hlx.page/

### CODE REVIEW

File: blocks/header/header.css [hunk 1]
```diff
@@ -78,143 +137,363 @@ header nav .nav-hamburger button {
...
-  height: 2px;
-  border-radius: 2px;
-  background: currentcolor;
+header nav[aria-expanded="true"] .nav-hamburger-icon{
+  background-image: url(../../icons/close.svg);
 }
 
-header nav[aria-expanded="false"] .nav-hamburger-icon::before {
-  top: -6px;
+/* secions */
```
Comment (AndreiAlexandruParaschiv) [inline blocks/header/header.css:151]: small typo: sections

File: blocks/header/header.css [hunk 2]
```diff
@@ -223,53 +502,250 @@ header nav .nav-sections ul > li > ul > li {
...
+    -webkit-box-pack: center;
+    -ms-flex-pack: center;
+    justify-content: space-between;
+    -webkit-box-align: center;
+    -ms-flex-align: center;
+    align-items: center;
+    padding: 0;
+    white-space: nowrap;
+    background-color: transparent;
+    flex-direction: row;
```
Comment (AndreiAlexandruParaschiv) [inline blocks/header/header.css:524]: duplicate of L514

File: blocks/header/header.css [hunk 3]
```diff
@@ -78,143 +137,363 @@ header nav .nav-hamburger button {
...
+    height: 100%;
+    margin: 0;
+    display: grid;
+    align-items: center;
+    grid-template: 
+                   'topbar' 32px
+                   'logo';
+  }
+  header nav .nav-brand div:first-child{
+    grid-area: 'topbar';
```
Comment (AndreiAlexandruParaschiv) [inline blocks/header/header.css:231]: grid-area: topbar and space before {

File: blocks/header/header.css [hunk 4]
```diff
@@ -1,68 +1,127 @@
...
+  border-bottom: 1px solid #ccc;
+  height: var(--nav-height);
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+}
+
+header nav a:hover{
+  text-decoration: none;
```
Comment (AndreiAlexandruParaschiv) [inline blocks/header/header.css:69]: space before { and add `text-shadow: -0.25px -0.25px 0 #0049B4, 0.25px 0.23px #0049B4;` to have the li elements bolded when hovering as the submenu

File: blocks/search/search.css
Comment (ekremney) [inline blocks/search/search.css:1]: font type/size are different for the search form placeholders:

Screen Shot 2023-05-05 at 19 47 59


Screen Shot 2023-05-05 at 19 47 19

File: blocks/header/header.js [hunk 5]
```diff
@@ -85,6 +100,156 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
...
+      if (currentActiveIndex > -1 && index === currentActiveIndex) {
+        item.classList.add('active');
+      } else {
+        item.classList.remove('active');
+      }
+    });
+  };
+
+  menuListItemCollection.forEach((item, index) => {
+    item.addEventListener('mouseenter', () => {
```
Comment (ekremney) [inline blocks/header/header.js:237]: if the search form is open when the `mouseenter` event fires, it should be closed:

Screen Shot 2023-05-05 at 19 48 25

File: blocks/header/header.css [hunk 6]
```diff
@@ -229,53 +522,244 @@ header nav .nav-sections ul > li > ul > li {
...
+    padding: 18px 10px;
+    width: auto;
   }
 
-  header nav .nav-sections > ul > li[aria-expanded="true"] > ul {
-    display: block;
+  header nav .nav-sections > ul > li > div.subnav-wrap ul > li a {
+    padding: 7px 0;
+    margin: 0 16px;
+    font-size: 16px;
```
Comment (ekremney) [inline blocks/header/header.css:562]: font-size for submenu items in tablet mode do not seem to be the same:

16px:
Screen Shot 2023-05-05 at 19 49 44

14px:
Screen Shot 2023-05-05 at 19 49 39

File: blocks/header/header.css [hunk 7]
```diff
@@ -229,53 +522,244 @@ header nav .nav-sections ul > li > ul > li {
...
     font-weight: 500;
+    overflow: hidden;
+    position: static;
+    max-height: unset;
   }
 
-  header nav .nav-sections > ul > li > ul {
-    display: none;
-    position: relative;
+  header nav .nav-sections ul > li > div.subnav-head a {
```
Comment (ekremney) [inline blocks/header/header.css:553]: in the original page, menu items has text decoration (bold) on hover:
Screen Shot 2023-05-05 at 20 38 00

File: blocks/header/header.css [hunk 8]
```diff
@@ -78,96 +143,178 @@ header nav .nav-hamburger button {
...
 header nav .nav-sections {
   grid-area: sections;
   flex: 1 1 auto;
-  display: none;
-  visibility: hidden;
+  display: block;
   background-color: var(--overlay-color);
 }
 
+header nav .nav-secondary-sections {
```
Comment (ekremney) [inline blocks/header/header.css:305]: not sure if I add this review to the correct line, but there seems to be a 1px wide whitespace on top of secondary grid item on tablet mode:

Screen Shot 2023-05-05 at 19 50 37

File: blocks/header/header.css [hunk 9]
```diff
@@ -78,96 +143,178 @@ header nav .nav-hamburger button {
...
+    margin: 0;
+    padding: 0 32px;
 }
 
-@media (min-width: 900px) {
-  header nav .nav-hamburger {
-    display: none;
-    visibility: hidden;
-  }
+header nav .nav-secondary-sections ul li a{
```
Comment (ekremney) [inline blocks/header/header.css:187]: we should add
```
letter-spacing: normal;
```
to this element. Not sure if this is correct place.

Screen Shot 2023-05-05 at 20 45 39

Screen Shot 2023-05-05 at 20 45 34

File: blocks/header/header.js [hunk 10]
```diff
@@ -85,6 +100,156 @@ function toggleMenu(nav, navSections, forceExpanded = null) {
   }
 }
 
+function adjustNavSection(closeNavSectionsCallback, navSection) {
+  const subNavUl = navSection.querySelector('ul');
+  if (subNavUl) {
+    const subNavToggle = createTag('div', { class: 'nav-toggle' });
+    subNavToggle.append(createTag('span', { class: 'icon icon-arrowright on' }));
+
+    subNavToggle.addEventListener('click', () => {
```
Comment (ekremney) [inline blocks/header/header.js:109]: there is an transition on menu closing in the original website:


https://user-images.githubusercontent.com/1171225/236543518-a589db1d-81cc-4692-99ca-6802aa1f368c.mov

**Review by ekremney — COMMENTED** [review]
In general it looks good. There's a few small issues that needs to be addressed

**Review by ekremney — APPROVED** [review]
Great work, LGTM!

**Pull request merged**